pkg/kubelet/logs/container_log_manager.go

```diff
@@ -213,66 +223,96 @@ func (c *containerLogManager) Clean(ctx context.Context, containerID string) err
 	return nil
 }
 
+func (c *containerLogManager) processQueueItems(worker int) {
+	klog.V(4).InfoS("Starting container log rotate worker", "workerID", worker)
+	for c.processContainer(context.Background(), worker) {
+	}
+	klog.V(4).InfoS("Terminating container log rotate worker", "workerID", worker)
+}
+
 func (c *containerLogManager) rotateLogs(ctx context.Context) error {
 	c.mutex.Lock()
 	defer c.mutex.Unlock()
+	klog.V(4).InfoS("Starting container log rotation sequence")
 	// TODO(#59998): Use kubelet pod cache.
 	containers, err := c.runtimeService.ListContainers(ctx, &runtimeapi.ContainerFilter{})
 	if err != nil {
 		return fmt.Errorf("failed to list containers: %v", err)
 	}
-	// NOTE(random-liu): Figure out whether we need to rotate container logs in parallel.
 	for _, container := range containers {
 		// Only rotate logs for running containers. Non-running containers won't
 		// generate new output, it doesn't make sense to keep an empty latest log.
 		if container.GetState() != runtimeapi.ContainerState_CONTAINER_RUNNING {
 			continue
 		}
-		id := container.GetId()
-		// Note that we should not block log rotate for an error of a single container.
-		resp, err := c.runtimeService.ContainerStatus(ctx, id, false)
-		if err != nil {
-			klog.ErrorS(err, "Failed to get container status", "containerID", id)
-			continue
-		}
-		if resp.GetStatus() == nil {
-			klog.ErrorS(err, "Container status is nil", "containerID", id)
-			continue
+		// Doing this to avoid additional overhead with logging of label like arguments that can prove costly
+		if v := klog.V(4); v.Enabled() {
```

Comment:
What's the difference between doing this and just doing `klog.V(4).InfoS("Starting container log rotation sequence")`, like you do above? Mostly just trying to understand as I'm not as familiar with `klog`. I assume that `klog.V(4).InfoS("...")` only logs if verbosity is `>= 4`, and this avoids building the string?

In other examples looks like if you explicitly make the check, you can remove the `V(4)`:

```
klog.InfoS("Adding new entry to the queue for processing", "id", container.GetId(), "name", container.Metadata.GetName(), "labels", container.GetLabels())
```

Reply:
This is a bit of a performance gain hack. The way klog serializers the data while using `InfoS` can lead to additional overhead if there are too many values. Doing the conditional check now means we do the klog part of workflow on only successful condition match for the verbosity.